### ISSUE TITLE
Crash when attempting to  import an OrgDb package from Bioconductor

### ISSUE THREAD

**Issue by lbeltrame** [rpy2/rpy2#710]
Minimal reproducer, with R 4.0.2 and the latest Bioconductor (observed on openSUSE Tumbleweed):

```
In [1]: from rpy2.robjects.packages import importr                                                                                                                                                                                                                            
In [2]: importr("org.Hs.eg.db")                                                                                                                                                                                                                                               
R[write to console]: Error in as.character(sys.call(sys.parent())[[1L]]) : 
  cannot coerce type 'closure' to vector of type 'character'

*** stack smashing detected ***: terminate
Aborted (core dumped)
```

**Comment by lgautier** [comment]
Do you know whether the issue appeared, when updating rpy2, or R/Bioconductor, or may has always been there ?

**Comment by lgautier** [comment]
This is odd.
The error occurs when calling the function `Rf_findVarInFrame` in the C API for the symbol `org.Hs.egPFAM` (it seems to just workd fine with all other symbols in the package before that).

**Comment by lgautier** [comment]
May be an R exception is raised before that symbols is reached, is not properly handled and R crashes down a little later without the root cause being apparent.

**Comment by lgautier** [comment]
Aha... the Bioconductor package is broken.

In R:

```r
> library(org.Hs.eg.db)
> org.Hs.eg.db::org.Hs.egPFAM
Error in as.character(sys.call(sys.parent())[[1L]]) : 
  cannot coerce type 'closure' to vector of type 'character'
```

What remains an issue with rpy2 is this is an error in R, but it leads to a segfault when calling R's `Rf_findVarInFrame`.

**Comment by lgautier** [comment]
The issue is reported on the Bioconductor side (https://github.com/Bioconductor/AnnotationForge/issues/15).

On the rpy2 side, this means that the function `Rf_findVarInFrame(symbol, environment)`, can surprisingly error and raise an exception even though `symbol` is defined in that environment.

**Comment by lbeltrame** [comment]
Thanks, I've subscribed to the Bioconductor issue as well.

**Comment by lgautier** [comment]
The content of the error message will be fixed in BioC, but this will still lead to a segfault. PR #717 is proposing a first fix but it is limited to the API mode (ABI will crash), and I am unsure there is a way to fix this in ABI mode with R's current C API.

**Comment by lgautier** [comment]
A fix in the main branch and will be in rpy2-3.4.0. Note that this is limited to the API mode though, and at the moment the compiling necessary for the API mode is not sorted out on Window.

**Comment by lgautier** [comment]
This was fixed a while ago (see comments above).
